pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_guardrails failed

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_guardrails [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10851685?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10851685?buildTab=artifacts#/) on release-23.1 @ [a7d63b0fcba4a8874b171ee58c1c951575659057](https://github.com/cockroachdb/cockroach/commits/a7d63b0fcba4a8874b171ee58c1c951575659057):


```
=== RUN   TestLogic_guardrails
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/35399f9f3f2a119316a70107be975c37/logTestLogic_guardrails252806416
    test_log_scope.go:79: use -show-logs to present logs inline
[15:23:42] setting distsql_workmem='36826B';
    logic.go:2880: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:34: SELECT * FROM guardrails
        expected:
        txn has read 51 rows, which is above the limit
        got:
        pq: txn has read 62 rows, which is above the limit: TxnID d1e322e5-f56a-4fcc-9e54-1397950f5d82 SessionID 17708aefb36041b30000000000000001
[15:23:42] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails with config fakedist: 9 tests, 1 failures
    logic.go:3992: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: error while processing
    logic.go:3992: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12446/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/guardrails:37: too many errors encountered, skipping the rest of the input
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/35399f9f3f2a119316a70107be975c37/logTestLogic_guardrails252806416
--- FAIL: TestLogic_guardrails (1.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_guardrails.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29605

--- a/pkg/ccl/logictestccl/tests/3node-tenant/generated_test.go
+++ b/pkg/ccl/logictestccl/tests/3node-tenant/generated_test.go
@@ -934,13 +934,6 @@ func TestTenantLogic_group_join(
 	runLogicTest(t, "group_join")
 }
 
-func TestTenantLogic_guardrails(
-	t *testing.T,
-) {
-	defer leaktest.AfterTest(t)()
-	runLogicTest(t, "guardrails")
-}
-
 func TestTenantLogic_hash_join(
 	t *testing.T,
 ) {

--- a/pkg/sql/logictest/testdata/logic_test/guardrails
+++ b/pkg/sql/logictest/testdata/logic_test/guardrails
@@ -1,6 +1,6 @@
-# LogicTest: !metamorphic-batch-sizes
-# We disable metamorphic batch sizes so that we read a predictable number of
-# rows in each limited scan.
+# LogicTest: !metamorphic-batch-sizes local local-legacy-schema-changer local-mixed-22.2-23.1 local-vec-off
+# We disable metamorphic batch sizes and only use local configs so that we read
+# a predictable number of rows in each limited scan.
 
 statement ok
 CREATE TABLE guardrails (i INT PRIMARY KEY);

--- a/pkg/sql/logictest/tests/fakedist-disk/generated_test.go
+++ b/pkg/sql/logictest/tests/fakedist-disk/generated_test.go
@@ -905,13 +905,6 @@ func TestLogic_group_join(
 	runLogicTest(t, "group_join")
 }
 
-func TestLogic_guardrails(
-	t *testing.T,
-) {
-	defer leaktest.AfterTest(t)()
-	runLogicTest(t, "guardrails")
-}
-
 func TestLogic_hash_join(
 	t *testing.T,
 ) {

--- a/pkg/sql/logictest/tests/fakedist-vec-off/generated_test.go
+++ b/pkg/sql/logictest/tests/fakedist-vec-off/generated_test.go
@@ -905,13 +905,6 @@ func TestLogic_group_join(
 	runLogicTest(t, "group_join")
 }
 
-func TestLogic_guardrails(
-	t *testing.T,
-) {
-	defer leaktest.AfterTest(t)()
-	runLogicTest(t, "guardrails")
-}
-
 func TestLogic_hash_join(
 	t *testing.T,
 ) {

--- a/pkg/sql/logictest/tests/fakedist/generated_test.go
+++ b/pkg/sql/logictest/tests/fakedist/generated_test.go
@@ -912,13 +912,6 @@ func TestLogic_group_join(
 	runLogicTest(t, "group_join")
 }
 
-func TestLogic_guardrails(
-	t *testing.T,
-) {
-	defer leaktest.AfterTest(t)()
-	runLogicTest(t, "guardrails")
-}
-
 func TestLogic_hash_join(
 	t *testing.T,
 ) {

